server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimGroupProvisioning.java

```diff
@@ -324,4 +349,12 @@ protected void validateOrderBy(String orderBy) throws IllegalArgumentException {
         super.validateOrderBy(orderBy, GROUP_FIELDS);
     }
 
+    private void validateAllowedUserGroups(String zoneId, ScimGroup group) {
```

Comment:
you want the validation in token creation ?

Reply:
I am not sure if that makes sense. That would require the validation to be done at every single token creation time, that feels wrong from the performance point of view. But it seems that you are correct, that it would solve this issue.

I was thinking of another idea: whenever the user config is updated, we need to revalidate all groups. That would make the config update more computationally heavy, but that is a lot less frequent than token creation.

Reply:
Ok, this would be a change in "Update Identity Zone", right? We would need to delete all groups which are not allowed anymore

Reply:
Added check: [56adc44](https://github.com/cloudfoundry/uaa/pull/2606/commits/56adc44dbf4b0281bf8044e852fce5ce3d60253a)